hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -366,7 +366,10 @@ object Label:
 object Scoped:
   def apply(syms: collection.Set[Local], body: Block): Block = body match
     case Scoped(syms2, body) =>
-      if syms2.isEmpty && syms.isEmpty then Scoped(Set.empty, body) else Scoped(syms ++ syms2, body)
+      if syms2.isEmpty && syms.isEmpty then Scoped(Set.empty, body)
+      else
+        assert(!syms2.exists(syms.contains), "overlapping symbols in nested Scoped")
```

Comment:
This looks uncomfortably expensive. Let's add a `validateIR: Bool` option in `Config` and only do the check when it's on.

Reply:
(Same below.)